node/pod.go

```diff
@@ -213,8 +213,11 @@ func (pc *PodController) enqueuePodStatusUpdate(ctx context.Context, q workqueue
 		if obj, ok := pc.knownPods.Load(key); ok {
 			kpod := obj.(*knownPod)
 			kpod.Lock()
+			defer kpod.Unlock()
+			if cmp.Equal(kpod.lastPodStatusReceivedFromProvider, pod) {
```

Comment:
Do we need to compare the entire pod, or just the pod status?

Reply:
I think it may be safer to compare the entire pod, which would not change the original behavior.